crates/bevy_transform/src/lib.rs

```diff
@@ -5,28 +5,42 @@
     html_favicon_url = "https://bevyengine.org/assets/icon.png"
 )]
 
+#[cfg(all(feature = "bevy_hierarchy", feature = "bevy_ecs"))]
 pub mod commands;
+
 /// The basic components of the transform crate
 pub mod components;
+
+#[cfg(all(feature = "bevy_hierarchy", feature = "bevy_ecs"))]
 pub mod helper;
+
 /// Systems responsible for transform propagation
+#[cfg(all(feature = "bevy_hierarchy", feature = "bevy_ecs"))]
 pub mod systems;
 
 #[doc(hidden)]
 pub mod prelude {
     #[doc(hidden)]
-    pub use crate::{
-        commands::BuildChildrenTransformExt, components::*, helper::TransformHelper,
-        TransformBundle, TransformPlugin, TransformPoint,
-    };
+    pub use crate::{components::*, TransformBundle, TransformPlugin, TransformPoint};
```

Comment:
Seems a bit weird leaving `TransformBundle` and `TransformPlugin` around, even if they're not of any use without their respective features being enabled.

Is there any reason?

Reply:
No reason- this PR was quickly put together to just ensure the strategy of reducing deps would work.

I was hoping for some feedback on the issue (https://github.com/bevyengine/bevy/issues/13529)
in order to determine if a PR would be accepted and how to go about it, e.g.

1. Like this current PR does it- as long as it compiles it's OK
2. Like what I think you're suggesting: Make _everything_ go away except the structs
3. Something else?

Converted to draft until I know more.